File: 0213/213.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var houseRobberIi = function(root) {
    if (!root) return 0;
    return 1 + Math.max(houseRobberIi(root.left), houseRobberIi(root.right));
};

File: 0213/213.go
func houseRobberIi(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := houseRobberIi(root.Left)
	r := houseRobberIi(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}